src/Octoshift/Handlers/DownloadLogsCommandHandler.cs

```diff
@@ -63,19 +63,14 @@ public async Task Handle(DownloadLogsCommandArgs args)
             _log.LogInformation($"MIGRATION LOG FILE: {args.MigrationLogFile}");
         }
 
-        args.MigrationLogFile ??= $"migration-log-{args.GithubOrg}-{args.GithubRepo}.log";
-
-        if (FileExists(args.MigrationLogFile))
+        // We check again below whether the file exists - but this is an extra check so we can fail early
+        // where the user has defined a custom filename, rather than using the auto-generated one
+        if (!args.Overwrite && FileExists(args.MigrationLogFile))
         {
-            if (!args.Overwrite)
-            {
-                throw new OctoshiftCliException($"File {args.MigrationLogFile} already exists!  Use --overwrite to overwrite this file.");
-            }
-
-            _log.LogWarning($"Overwriting {args.MigrationLogFile} due to --overwrite option.");
+            throw new OctoshiftCliException($"File {args.MigrationLogFile} already exists! Use --overwrite to overwrite this file.");
         }
 
-        var result = await _retryPolicy.RetryOnResult(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), string.Empty,
+        var result = await _retryPolicy.RetryOnResult<(string MigrationLogUrl, string MigrationId)?>(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), r => r?.MigrationLogUrl.IsNullOrWhiteSpace() ?? false,
```

Comment:
```suggestion
        var result = await _retryPolicy.RetryOnResult(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), r => r?.MigrationLogUrl.IsNullOrWhiteSpace() ?? false,
```

the compiler can infer the generic type here